### PR TITLE
Reorganize reusable singleton wallet code

### PR DESCRIPTION
**Pull request by aqk**
No change, no user visible change.

### CODE REVIEW

**Comment by aqk** [comment]
This reorg is to support sharing of code between Singleton wallets, in preparation for the DAO Wallet.

File: chia/wallet/did_wallet/did_wallet_puzzles.py [hunk 1]
```diff
@@ -114,15 +90,6 @@ def is_did_innerpuz(inner_f: Program) -> bool:
     return inner_f == DID_INNERPUZ_MOD
 
 
-def is_did_core(inner_f: Program) -> bool:
```
Comment (matt-o-how) [inline chia/wallet/did_wallet/did_wallet_puzzles.py:117]: Glad to see the back of this function.
Once upon a time the singleton layer was called the did core, which is why this function is called that. 
Hopefully that's the last of references to did_core

File: chia/wallet/did_wallet/did_wallet.py [hunk 2]
```diff
@@ -10,6 +10,7 @@
 
 from blspy import AugSchemeMPL, G1Element, G2Element
 
+import chia.wallet.singleton
```
Comment (Quexington) [inline chia/wallet/did_wallet/did_wallet.py:13]: I'm not a huge fan of this, can we either throw the other singleton functions that we need into the import below or import this `as singleton`?

**Review by Quexington — APPROVED** [review]
Stamping this with approval because it's an improvement and fine to merge, but I will note that it's not entirely complete as we also have chia.wallet.puzzles.singleton_top_layer_v1_1.py which is the same as this file but with more and different functions.  Moving those functions in here might be a good idea as well.

**Review by matt-o-how — APPROVED** [review]
Good changes.
Looks good to me.

File: chia/wallet/singleton.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
+from __future__ import annotations
+
+from typing import Optional
+
+from chia.types.blockchain_format.program import Program
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.wallet.puzzles.load_clvm import load_clvm_maybe_recompile
+from chia.wallet.util.curry_and_treehash import calculate_hash_of_quoted_mod_hash, curry_and_treehash
+
+SINGLETON_TOP_LAYER_MOD = load_clvm_maybe_recompile("singleton_top_layer_v1_1.clvm")
```
[the remark this answers is not in the document]
Comment (aqk) [reply chia/wallet/singleton.py:10]: Yes, I can move that here, too

File: chia/wallet/did_wallet/did_wallet.py (hunk 2 above)
Comment (aqk) [reply chia/wallet/did_wallet/did_wallet.py:13]: Oh, sure.

**Comment by aqk** [comment]
> Stamping this with approval because it's an improvement and fine to merge, but I will note that it's not entirely complete as we also have chia.wallet.puzzles.singleton_top_layer_v1_1.py which is the same as this file but with more and different functions. Moving those functions in here might be a good idea as well.

Thanks, I'll open a new PR for that

**Review by emlowe — APPROVED** [review]

**Pull request merged**